addons/hide-ads/userscript.js

```diff
@@ -0,0 +1,201 @@
+  const advertisingUser = "[User's messages blocked]";
+  const advertisingSpam = "[Marked as general spam]";
+
+  function filterSpamComments(content) {
+    //These are just made by chatGPT after giving it a lot of example comments
+    const repeatedCharactersRegex = /(.)\1{20,}/;
+    const meaninglessRegex = /^[\W\s]*$/;
+
+    if (repeatedCharactersRegex.test(content)) {
+      return true;
+    }
+
+    if (meaninglessRegex.test(content)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  function handleDisabled(commentContent, profileReplyButton, comment, handleProfileReplyButton, blockType) {
+    commentContent.innerHTML = idToContent[commentContent.closest(".comment").getAttribute("id")];
+    commentContent.classList.remove("advertising");
+    comment.classList.remove("contains-advertising");
+
+    if (handleProfileReplyButton()) {
+      profileReplyButton.style.display = "inline";
+    }
+
+    commentContent.style.cursor = "";
+  }
+
+  function handleReenabled(commentContent, profileReplyButton, comment, handleProfileReplyButton, blockType) {
+    commentContent.innerHTML =
+      blockType === "content" ? advertisingContent : blockType === "user" ? advertisingUser : advertisingSpam;
+    commentContent.classList.add("advertising");
+    comment.classList.add("contains-advertising");
+
+    if (handleProfileReplyButton()) {
+      profileReplyButton.style.display = "";
+    }
+
+    commentContent.style.cursor = "pointer";
+  }
+
+  let settings = {};
+
+  function getNewSettings() {
+    settings["projects"] = addon.settings.get("projects");
+    settings["studios"] = addon.settings.get("studios");
+    settings["profiles"] = addon.settings.get("profiles");
+    settings["method"] = addon.settings.get("method");
+    settings["spamFilter"] = addon.settings.get("spamFilter");
+    settings["matches"] = addon.settings.get("matches");
+  }
+
+  getNewSettings();
+
+  function handleComment(element, blockType) {
+    if (addon.settings.get("method") === "hide") {
+      element.style.display = "none";
+    } else {
+      const comment = element.classList.contains("comment") ? element : element.querySelector(".comment");
+      comment.classList.add("contains-advertising");
+
+      let commentContent;
+      let profileReplyButton;
+
+      if (url.includes(settingsToUrl["profiles"])) {
+        commentContent = element.querySelector(".info>.content");
+        profileReplyButton = commentContent.nextElementSibling.lastElementChild;
+      } else {
+        commentContent = element.querySelector(".comment-content");
+      }
+
+      function handleProfileReplyButton() {
+        if (profileReplyButton?.classList.contains("reply") && url.includes(settingsToUrl["profiles"])) {
+          return true;
+        } else return false;
+      }
+
+      const key = comment.getAttribute("id");
+      const value = commentContent.innerHTML;
+
+      idToContent[key] = value;
+
+      commentContent.innerHTML =
```

Comment:
This should probably use `textContent` - this is currently not an issue because the messages are hardcoded, but that's not the case if they are translated.

Reply:
I can not do that easily without messing up emojis, this was originally using text content but i have to go with this method to ensure the comment's content stays the exact same

Reply:
`innerHTML` is fine when inserting the actual comment I think, but I mean using `textContent` for the [Marked as ...] messages.

Reply:
oh do i do that? uhmmm yeah ill change it